src/RichXMLParser.m

```diff
@@ -57,8 +57,15 @@ -(BOOL)parseRichXML:(NSData *)xmlData
         xmlDocument = [[NSXMLDocument alloc] initWithData:xmlData
                                                         options:NSXMLNodeOptionsNone
                                                         error:&error];
-        if (error) {
-            if (error.code == 73) return NO;
+        if (xmlDocument == nil && error != nil) {
+			if ([error.domain isEqualToString:NSXMLParserErrorDomain]) {
+				if (error.code == NSXMLParserGTRequiredError)
+					return NO;
+				if (error.code == NSXMLParserPrematureDocumentEndError)
+					return NO;
+			}
+			// Do we even want to try this?
```

Comment:
The second attempt with the option `NSXMLDocumentTidyXML` is really useful : it will retrieve useful info from many incorrect feeds. See this Apple example : https://developer.apple.com/library/mac/documentation/Cocoa/Conceptual/NSXML_Concepts/Articles/CreatingXMLDoc.html